subsys/net/ip/net_if.c

```diff
@@ -4196,6 +4205,65 @@ int net_if_up(struct net_if *iface)
 	return status;
 }
 
+int net_if_connect(struct net_if *iface)
+{
+	int status = 0;
+
+	NET_DBG("iface %p connect", iface);
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	if (!net_if_flag_is_set(iface, NET_IF_UP)) {
+		status = -ESHUTDOWN;
+		goto out;
+	}
+
+	/* Question to reviewers:
+	 * Would it be cleaner to attach the underlying API implementation to the
+	 * net_l2 struct instead...? This would require modifications be made to allow
+	 * offloaded sockets to have L2 implementations. But I think we will need this
```

Comment:
I agree, let's do both the offloaded net_l2 dummy, and your `net_l2_connectivity` suggestion. I guess this would be the right PR to get started on the dummy, since we know at this stage we'll want it down the line.